src/main/java/org/eln2/mc/common/CommonEvents.kt

```diff
@@ -1,17 +1,30 @@
 package org.eln2.mc.common
 
-import net.minecraftforge.event.server.ServerStartedEvent
+import net.minecraft.server.level.ServerLevel
+import net.minecraftforge.event.ServerChatEvent
+import net.minecraftforge.event.TickEvent
 import net.minecraftforge.eventbus.api.SubscribeEvent
 import net.minecraftforge.fml.common.Mod
+import net.minecraftforge.server.ServerLifecycleHooks
+import org.eln2.mc.Eln2
+import org.eln2.mc.common.cell.CellGraphManager
 
 @Mod.EventBusSubscriber
 object CommonEvents {
     @SubscribeEvent
-    fun onServerStarted(event : ServerStartedEvent) {
-        // we can now start the electrical simulator
-        event.server.allLevels.forEach{
-            /*val nodeManager = it.getNodeManager()
-            nodeManager.prepare()*/
+    fun onServerTick(event : TickEvent.ServerTickEvent){
+        if(event.phase == TickEvent.Phase.END){
+            ServerLifecycleHooks.getCurrentServer().allLevels.forEach{
+                CellGraphManager.getFor(it).tickAll()
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onChat(event : ServerChatEvent){
+        Eln2.LOGGER.info("Chat: ${event.message}")
+        if(event.message == "build"){
+            CellGraphManager.getFor(event.player.level as ServerLevel).graphs.values.forEach{ it.build() }
         }
     }
 }
```

Comment:
What is the purpose of this exactly? I ran the command and it doesn't seem to do anything too useful? Probably for debugging?

Reply:
It was used before I rebuild the circuits on each edit. It can be removed.